### ISSUE TITLE
TF 2.3 - loading of saved_model from disk with ragged=True input is slower than ragged=False

### ISSUE THREAD

**Issue by michalderdak** [tensorflow/tensorflow#43263]
**System information**
Google Colab Notebook with TF 2.3

**Describe the current behavior**
Loading saved model with input `tf.keras.layers.Input(shape=[None], dtype=tf.int64, ragged=True)` is 5-10x slower than `tf.keras.layers.Input(shape=[None], dtype=tf.int64, ragged=False)`

**Describe the expected behavior**
The above should not impact performance

**Standalone code to reproduce the issue**
https://colab.research.google.com/drive/10ZXvdbp1lf8X2LLtemcHr2rX692n2jv9?usp=sharing

**Other info / logs**
The above is achieved with a custom model implementation. However, when I used the model from one of the Tensorflow tutorials I could not replicate the issue.

On top of that, we were able to replicate this issue with tensorflow serving where loading of the saved_model + first prediction was 5-10x slower than model without ragged tensors.

**Comment by amahendrakar** [comment]
Was able to reproduce the issue with [TF v2.3](https://colab.research.google.com/gist/amahendrakar/f9e28ebe24fb701b7859b6c3d56b9bdb/43263.ipynb) and [TF-nightly](https://colab.research.google.com/gist/amahendrakar/89196ab681b8f0f3fe32419338e8328d/43263-tf-nightly.ipynb). 

No time difference when running TensorFlow example, slows down when running a custom model implementation. Please find the attached gist. Thanks!

**Comment by tanguycdls** [comment]
@michalderdak did you have any luck on this ? we're still struggling with it in our side... thanks

**Comment by michalderdak** [comment]
@chunduriv I've replicated the issue with `TF2.6` and `tf-nightly`. Furthermore, I have added examples without our custom loss function to have as little custom code as possible.

see the notebooks here:
https://colab.research.google.com/drive/1FW2cVAGSL5aWuFBbDcLNbwvtseOaOYO2?usp=sharing
https://colab.research.google.com/drive/1YNKSG21CQf3Q6mQFzrLKNO6FzB7Z6j2T?usp=sharing

**Comment by tilakrayal** [comment]
Hi,

Thank you for opening this issue. Since this issue has been open for a long time, the code/debug information for this issue may not be relevant with the current state of the code base.

The Tensorflow team is constantly improving the framework by fixing bugs and adding new features. We suggest you try the latest TensorFlow [version](https://pypi.org/project/tensorflow/) with the latest compatible hardware configuration which could potentially resolve the issue. If you are still facing the issue, please create a new GitHub issue with your latest findings, with all the debugging information which could help us investigate.

Please follow the [release notes](https://github.com/tensorflow/tensorflow/blob/master/RELEASE.md) to stay up to date with the latest developments which are happening in the Tensorflow space.
